src/distilabel/steps/generators/huggingface.py

```diff
@@ -136,7 +154,7 @@ def process(self, offset: int = 0) -> "GeneratorStepOutput":
             transformed_batch = self._transform_batch(batch)
             batch_size = len(transformed_batch)
             num_returned_rows += batch_size
```

Comment:
If the comment below is the case i.e. we can return more data that the specified data, shouldn't we count all the total rows, and on each iter `if num_returned_rows > self.num_examples` then `transformed_batch[:(self.num_examples - num_returned_rows)]`?

Meaning that if we have say 10 rows and we use batches of size 5, but `num_examples` is set to 9, then we batch that so that we return 5 and then 5 too, since we're not checking that the total number of rows is bigger than `num_examples` right?

Reply:
I changed the condition below because it makes more sense to have `>=` there, we don't have the problem anymore because once filtered the dataset even if we wanted to retrieve more rows, the dataset already has 9 examples instead of 10, so we won't yield more